libcxx/include/__vector/vector_bool.h

```diff
@@ -398,18 +398,12 @@ class _LIBCPP_TEMPLATE_VIS vector<bool, _Allocator> {
   template <class _InputIterator, class _Sentinel>
   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 void
   __init_with_sentinel(_InputIterator __first, _Sentinel __last) {
-#if _LIBCPP_HAS_EXCEPTIONS
-    try {
-#endif // _LIBCPP_HAS_EXCEPTIONS
-      for (; __first != __last; ++__first)
-        push_back(*__first);
-#if _LIBCPP_HAS_EXCEPTIONS
-    } catch (...) {
-      if (__begin_ != nullptr)
-        __storage_traits::deallocate(__alloc(), __begin_, __cap());
-      throw;
-    }
-#endif // _LIBCPP_HAS_EXCEPTIONS
+    auto __guard = std::__make_exception_guard(__destroy_vector(*this));
```

Comment:
IIUC, this isn't fixing a bug, right? I think both versions are 100% equivalent.

If that's not the case, then what is the behaviour change? We should have a test for it.

Reply:
I think they are 100% equivalent. Currently, there are no exception tests for vector<bool> although we have exceptions tests for vector<T>. I can add one.

Reply:
If you're able to add one, that would be amazing.

Reply:
I have added an exception test file for `vector<bool>`, which covers all constructors that might raise exceptions in various scenarios. Since the exception tests for `vector<bool>` and `vector<T>` are similar, I factored out the common code used by both tests into a header file `exception_test_helpers.h`. 

Additionally, I found some issues with the existing exceptions tests for `vector<T>`: some tests did not throw exception as intended, making them not useful, and some tests did not throw exceptions at the right points, failing to fulfill their purpose. We may need another PR to fix that.